fullscreen.bs

```diff
@@ -340,23 +330,26 @@ attribute's getter must run these steps:
   <p>As part of the next <a>animation frame task</a>, run these substeps:
 
   <ol>
+   <li><p>If <var>doc</var>'s <a>fullscreen element</a> is null, then resolve <var>promise</var>
+   with undefined and terminate these steps.
 
    <li><p>Let <var>exitDocs</var> be the result of
    <a lt="collect documents to unfullscreen">collecting documents to unfullscreen</a> given
-   <var>doc</var>.
+   <var>doc</var> and the <var>fully flag</var>.
    <!-- cross-process -->
 
-   <li><p>If <var>resize</var> is true and <var>topLevelDoc</var> is not in <var>exitDocs</var>,
-   <a>fully exit fullscreen</a> <var>topLevelDoc</var>, reject <var>promise</var> with a
-   <code>TypeError</code> exception, and terminate these steps.
-
    <li><p>If <var>exitDocs</var> is the empty set, append <var>doc</var> to <var>exitDocs</var>.
 
    <li><p>If <var>exitDocs</var>'s last <a for=/>document</a> has a
    <a>browsing context container</a>, append that <a>browsing context container</a>'s
    <a>node document</a> to <var>exitDocs</var>.
    <!-- cross-process -->
 
+   <li><p>If <var>resize</var> is true, then: while <var>exitDocs</var>'s last <a for=/>document</a>
+   has a <a>browsing context container</a>, append that <a>browsing context container</a>'s
+   <a>node document</a> to <var>exitDocs</var>.
+   <!-- cross-process -->
+
    <li><p>Let <var>descendantDocs</var> be an ordered set consisting of <var>doc</var>'s
```

Comment:
Using *doc* here seems like a problem before and after this change. Before, the fullscreen element in one of *exitDocs* may have been another iframe, since we have no hierarchy restrictions now. The wrong part of the frame tree is then going to get its fullscreen state cleared.

With this change, there's the additional problem that if *fully flag* is set, we'll exit fully in all those documents, and there may be a bunch of frames that need state cleared that don't get it.

I'm thinking that maybe the synchronous intro of "exit fullscreen" should instead determine a single *exitDoc*, from which one or all fullscreen top layer elements will be removed depending on the *fully flag*. Then there will be no "collecting documents to unfullscreen" in the animation frame task, only an attempt to make it so. Would that kind of model work? Should I work it into this PR or not?